pkg/manifests/nginx_test.go

```diff
@@ -455,17 +455,11 @@ func TestIngressControllerResources(t *testing.T) {
 	}
 
 	for _, tc := range controllerTestCases {
-		for _, version := range nginxVersions {
-			t.Run(
-				func() string {
-					if version == nil {
-						return tc.Name + "-unspecified"
-					}
-					return tc.Name + version.name
-				}(),
-				func(t *testing.T) {
-					t.Parallel()
-
+		tc := tc
+		t.Run(
```

Comment:
this was a race condition detected by race condition unit test flag

Reply:
I do not know why this wasn't discovered by our github pipeline because that also uses the race flag.
https://github.com/Azure/aks-app-routing-operator/blob/5109e8158b8136eb1250134516e537f025c343c1/.github/workflows/unit.yaml#L51